src/csexp_rpc/csexp_rpc.mli

```diff
@@ -0,0 +1,63 @@
+(** Threaded RPC Client & Server for Csexp payloads. This RPC implementation is
+    entirely structure free and it is up to users of this library to impose some
+    sort of conventions on the protocol.
+
+    On Unix/Mac, RPC uses Unix domain sockets. On Windows, named pipes are used.
+
+    All functions reurning a fiber are non-blocking and run in their own thread. *)
+open Stdune
+
+module Scheduler : sig
+  (** Hook into the fiber scheduler. *)
+  type t =
+    { on_event : Fiber.fill -> unit  (** Called when any rpc event completes *)
+    ; register_pending_ivar : unit -> unit
+          (** Called before any ivar is created *)
+    ; spawn_thread : (unit -> unit) -> unit
+          (** We spawn threads through this function in case the scheduler wants
+              to block signals *)
+    }
+end
+
+module Session : sig
+  (** Rpc session backed by two threads. One thread for reading, and another for
+      writing *)
+  type t
+
+  val create : in_channel -> out_channel -> Scheduler.t -> t
+
+  (* [write t x] writes the s-expression when [x] is [Some sexp], and closes the
+     session if [x = None ] *)
+  val write : t -> Sexp.t option -> unit Fiber.t
+
+  (** If [read] returns [None], the session is closed and all subsequent reads
+      will return [None] *)
+  val read : t -> Sexp.t option Fiber.t
+end
+
+type where =
+  [ `Unix of Path.t
+  | `Ip of [ `Ipv4 | `Ipv6 ] * Unix.inet_addr * [ `Port of int ]
+  ]
```

Comment:
Why use a polymorphic variant rather than a plan variant here?

Reply:
Because the same type is used in both `Dune_rpc` and `Csexp_rpc` and I didn't want to introduce a dependency, and was a bit lazy to add a conversion step.

Reply:
Moved to #4237

Reply:
Ack. BTW, we could also just pass a `Unix.sockaddr` to the various function in this module and leave the more detailed type for the other library. All the information is already contained in a `Unix.sockaddr`, in particular you can get the domain with `Unix.domain_of_sockaddr`

Reply:
Okay, will give this a try.